targets/TARGET_STM/sleep.c

```diff
@@ -40,17 +40,25 @@ extern void HAL_ResumeTick(void);
 
 void hal_sleep(void)
 {
+    // Disable IRQs
+    core_util_critical_section_enter();
```

Comment:
Can you confirm this is a change in behaviour since the previous RTX? And why the distinction between debug and non-debug builds?

We'll need someone more familiar with the RTX port to comment. My analysis above was assuming a dedicated "idle" handler in an OS scheduler - it may be somewhat different if hal_sleep is just called as "normal" code in a minimum-priority idle thread.